server/consumer.go

```diff
@@ -2529,6 +2534,16 @@ func (o *consumer) addAckReply(sseq uint64, reply string) {
 	o.replies[sseq] = reply
 }
 
+// Used to remember messages that need to be sent for a replicated consumer, after delivered quorum.
+// Lock should be held.
+func (o *consumer) addReplicatedQueuedMsg(pmsg *jsPubMsg) {
+	// Is not explicitly limited in size, but will at maximum hold maximum ack pending.
```

Comment:
Are we sure this comment is true?

Reply:
The stream sequence is used as a key, which is the same for `o.pending`. That means it can only ever become as large as max length of `o.pending`, i.e. `MaxAckPending`.

Reply:
ok so when new message comes into stream and we kick the consumer we just bail because we hit max pending yes?

Reply:
`o.getNextMsg` bails based on hitting max pending, indeed.

Reply:
Understood but I am wondering if we should short circuit in stream's signalConsumers()? WDYT?

Reply:
Could potentially be an additional optimization, although can't say how large the gain would be.
Either way I think it should be in a different PR.

Reply:
Let's do that PR after this one that short circuits.